service/storage_service.cc

```diff
@@ -2079,6 +2084,19 @@ future<> storage_service::raft_state_monitor_fiber(raft::server& raft, cdc::gene
 
 
 future<> storage_service::raft_replace(raft::server& raft_server, raft::server_id replaced_id, gms::inet_address replaced_ip) {
+    auto ignore_nodes_strs = utils::split_comma_separated_list(_db.local().get_config().ignore_dead_nodes_for_replace());
+    std::unordered_set<raft::server_id> ignored_ids;
+    for (const sstring& n : ignore_nodes_strs) {
+        raft::server_id ignored_id;
+        // FIXME: IPs should be supported, but the current implementation of raft_address_map doesn't allow this.
```

Comment:
Sure, but I'd like to know if there is any expected solution for this FIXMEs to provide more information in the issue ticket. Is it simply adding a new member function -- `find_by_inet_address` -- to `raft_address_map` together with some unit tests?

By the way, I left two TODOs in `test_raft_ignore_nodes.py` that depend on these FIXMEs and should be done together with them.

Reply:
> Is it simply adding a new member function -- find_by_inet_address -- to raft_address_map together with some unit tests?

That is one of the alternatives.
Although we should probably do the translation the same way regular removenode does, IIRC it's done using `locator::topology` somehow.

Another alternative is to update the documentation that we no longer support passing IP addresses to this option. However that's a bit problematic because there was no deprecation period.

Best would be a mix of both: update the documentation that passing IPs is deprecated, and do the translation here. Two releases later we would update the documentation that passing IPs is no longer supported and remove the translation.

Reply:
I created the follow-up issue -- #15126.